store/postgres/src/deployment.rs

```diff
@@ -897,8 +898,8 @@ pub fn create_deployment(
         d::non_fatal_errors.eq::<Vec<String>>(vec![]),
         d::earliest_ethereum_block_hash.eq(b(&earliest_block)),
         d::earliest_ethereum_block_number.eq(n(&earliest_block)),
-        d::latest_ethereum_block_hash.eq(sql("null")),
-        d::latest_ethereum_block_number.eq(sql("null")),
+        d::latest_ethereum_block_hash.eq(b(&latest_block)),
+        d::latest_ethereum_block_number.eq(n(&latest_block)),
```

Comment:
That will screw up grafting; grafting has to happen when the subgraph is started, and a `null` latest block is used as a signal for `DeploymentStore.start_subgraph` that the subgraph still needs more initialization, which includes the possibly long time for grafting. What we should do instead is change the logic in `DeploymentStore.start_subgraph` to:
```
if has_pending_graft() {
  do_the_graft()
}
// make sure we see the effect of grafting
let latest_block = get_latest_block_from_db();
if latest_block.is_none() {
  set latest_block = earliest_block
}
```

Essentially, the logic for initializing the latest block ptr is: if the subgraph is grafted, use that, otherwise use the earliest block.

Reply:
@lutter I think that should work, yes. Then we can drop the `latest_block` on `DeploymentCreate` again. And ideally, also add a comment to this new `// make sure we see the effect of grafting` section that in the case of subgraphs with a `startBlock`, the subgraphs are initialized with `earliestBlock = min(startBlocks)` and setting the latest block here makes that take effect before starting up.